apps/opik-frontend/src/api/vllm/useVllmModels.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  PROVIDER_MODEL_TYPE,
+  PROVIDER_MODELS_TYPE,
+  PROVIDER_TYPE,
+} from "@/types/providers";
+import { useQuery } from "@tanstack/react-query";
+import axios from "axios";
+
+const fetchVllmModels = async (base_url: string) => {
+  // Use axios to fetch instead of api, since api is withCredentials, and vllm server usually would have CORS response *
+  // You cannot have withCredentials true with the server returning Access-Control-Allow-Origin as *
+  const { data } = await axios.get(`${base_url}/models`);
+  const models: string[] = data.data.map((item: { id: string }) => item.id);
+  let retVal: Partial<PROVIDER_MODELS_TYPE> | undefined = undefined;
+  if (models.length > 0) {
+    retVal = {
+      [PROVIDER_TYPE.VLLM]: models.map((m) => ({
+        value: m.trim() as PROVIDER_MODEL_TYPE,
+        label: m.split("/").pop()!,
+      })),
+    };
+  }
+  return retVal;
+};
```

Comment:
I would go with manually typing in the model name as a first step. If the model is run locally, this forces the user to use `localhost` as the base URL. However, when the backend then tries to make this call it fails because this request is made from within a docker container, in which case the base URL should be set to `host.docker.internal` or similar, or the entire thing fails.